### PR TITLE
Add matching logic

### PR DESCRIPTION
**Pull request by ashellunts**

### CODE REVIEW

File: SetGame.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -7,13 +7,16 @@
...
 		1978206826AAC30700624E75 /* SceneDelegate.swift in Sources */ = {isa = PBXBuildFile; fileRef = 1978206726AAC30700624E75 /* SceneDelegate.swift */; };
 		1978206A26AAC30700624E75 /* ViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 1978206926AAC30700624E75 /* ViewController.swift */; };
 		1978206D26AAC30700624E75 /* Main.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = 1978206B26AAC30700624E75 /* Main.storyboard */; };
 		1978206F26AAC30700624E75 /* Assets.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = 1978206E26AAC30700624E75 /* Assets.xcassets */; };
 		1978207226AAC30700624E75 /* LaunchScreen.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = 1978207026AAC30700624E75 /* LaunchScreen.storyboard */; };
-		1978207D26AAC30700624E75 /* SetGameTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 1978207C26AAC30700624E75 /* SetGameTests.swift */; };
+		1978207D26AAC30700624E75 /* SetGameModelTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 1978207C26AAC30700624E75 /* SetGameModelTests.swift */; };
```
Comment (izimenkov) [inline SetGame.xcodeproj/project.pbxproj:19]: Wouldn't it be a good idea to have all the renamings in a separate commit just to easify the reviews?

File: SetGame/Card.swift [hunk 2]
```diff
@@ -14,10 +14,10 @@ func randomEnum(_ values: [Any]) -> Any {
 
 enum Shape {
 	case circle
-	case squar
```
Comment (izimenkov) [inline SetGame/Card.swift:17]: Probably it's also a good idea to fix the typos in a separate commit to easify the reviews?

File: SetGame/CardsMatching.swift [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import Foundation
+
+func valuesMatching<Type: Equatable>(_ value1: Type, _ value2: Type, _ value3: Type) -> Bool {
+	let allSameShape = value1 == value2 && value2 == value3
+	let allDifferentShape = value1 != value2 && value2 != value3 && value3 != value1
+
+	return allSameShape || allDifferentShape
+}
+
+func CardsMatching(_ card1: Card, _ card2: Card, _ card3: Card) -> Bool {
```
Comment (izimenkov) [inline SetGame/CardsMatching.swift:17]: Here one can read the rules: https://en.wikipedia.org/wiki/Set_(card_game).
I have not seen the word "match".
Probably one should find a better name for this from the domain language of the game.
Something with some words: "these three cards form a set".

File: SetGame/ViewController.swift [hunk 4]
```diff
@@ -39,6 +39,7 @@ class ViewController: UIViewController {
 	func updateUI() {
 		for i in vm.cardsOnTable.indices {
 			buttons[i].isHidden = false
+			buttons[i].setTitle(vm.cardsOnTable[i].text, for: .normal)
```
Comment (izimenkov) [inline SetGame/ViewController.swift:42]: It could also be in a separate commit. It does not fit the current commit message!

File: SetGameTests/CardsMatchingTest.swift [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+//  CardsMatchingTest.swift
+//  SetGameTests
+//
+//  Created by Artur Shellunts on 12.09.21.
+//
+
+import XCTest
+@testable import SetGame
+
+class CardsMatchingTest : XCTestCase {
```
Comment (izimenkov) [inline SetGameTests/CardsMatchingTest.swift:11]: Perfect!

File: SetGameTests/ValuesMatchingTest.swift [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+//  CardsMatchingTest.swift
+//  SetGameTests
+//
+//  Created by Artur Shellunts on 12.09.21.
+//
+
+import XCTest
+@testable import SetGame
+
+class ValuesMatchingTest : XCTestCase {
```
Comment (izimenkov) [inline SetGameTests/ValuesMatchingTest.swift:11]: Consider what Jan said. It's probably an implementation detail of the "CardsMatching" strategy and shouldn't be tested separately.

File: SetGameTests/SetGameModelTests.swift [hunk 7]
```diff
@@ -1,21 +1,21 @@
...
 
 import XCTest
 @testable import SetGame
 
-class SetGameTests: XCTestCase {
+class SetGameModelTests: XCTestCase {
 
 	var cardsOnTable: Int!
 	var deckSize: Int!
 	var vm: SetGameModel!
```
Comment (izimenkov) [inline SetGameTests/SetGameModelTests.swift:15]: What does "vm" mean?

File: SetGame/SetGameModel.swift [hunk 8]
```diff
@@ -31,16 +31,26 @@ class SetGameModel {
 		checkMatching()
 	}
 
+	//TODO test
```
Comment (izimenkov) [inline SetGame/SetGameModel.swift:34]: Would it be reasonable to have at least a smoke test for this change?

**Review by izimenkov — APPROVED** [review]

File: SetGameTests/SetGameModelTests.swift (hunk 7 above)
Comment (ashellunts) [reply SetGameTests/SetGameModelTests.swift:15]: view model

File: SetGame/SetGameModel.swift (hunk 8 above)
Comment (ashellunts) [reply SetGame/SetGameModel.swift:34]: sure
